web/assets/js/components/gameboard.ts

```diff
@@ -148,18 +158,38 @@ const Gameboard = class extends Component {
     }
 
     //game turn update
+
     if (this.color === this.chess.turn()) {
-      this.call('stopTimer', [init], 'gameplayers', 'rival');
-      this.call('startTimer', [init], 'gameplayers', 'me');
       this.allowedToMove = true;
+      this.call('stopTimer', [init], 'gameplayers', 'rival');
+
+      if (this.initMove) {
+        this.initMove = false;
+        this.initMoveTimer = setTimeout(() => {
+          console.warn('no 1 move');
+          //TODO: call action claimTimeout @Miloš / @Morgan
+        }, 30000); //30sec first move
+      } else {
+        this.call('startTimer', [init], 'gameplayers', 'me');
+      }
     } else {
+      clearTimeout(this.initMoveTimer);
       this.call('stopTimer', [init], 'gameplayers', 'me');
       this.call('startTimer', [init], 'gameplayers', 'rival');
       this.allowedToMove = false;
       this.rivalMove();
     }
   }
 
+  private async intervalCheckForTimeout() {
+    const actions: Actions = await Actions.getInstance();
+    this.checkTimeoutTimer = setInterval(async () => {
+      const timeoutState = await actions.getTimeoutState(this.gameId);
+
+      if (timeoutState) this.call(this.engine(false, 'timeout'));
```

Comment:
Note that this can potentially be called multiple times (we need to make sure this interval is properly disabled after it triggers a timeout)

Reply:
Yes I draw it very quickly and I wanted to check it. Interval + promise are hard to manage if we forgot closing them properly.